src/session.ts

```diff
@@ -46,6 +49,11 @@ export class Session extends Client implements ErdstallSession {
 		this.enclaveWriter.on("error", () => {
 			this.nonce = 0n;
 		});
+		this.receiptDispatcher = new ReceiptDispatcher(this.erdstallConn);
+		this.on(
+			"receipt",
+			this.receiptDispatcher.watch.bind(this.receiptDispatcher),
+		);
```

Comment:
We still have to brainstorm and think if we come up with a better architecture that does not require us to harden our logic against a users desire to clean stuff up...